modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/SplitCommerceOrderHelperImpl.java

```diff
@@ -59,6 +66,55 @@ public void splitByCatalog(long commerceOrderId) throws Exception {
 
 		if (numberCommerceOrder > 1) {
 			_createSupplierOrders(customerCommerceOrder, commerceOrderItemMap);
+			_handleBookedQuantity(commerceOrderId);
```

Comment:
I don't see this being an "atomic" transaction. What if the customer's bq are restocked and then something breaks? I will be in a incoerent status.